pkg/interpolatorwriter/writer.go

```diff
@@ -0,0 +1,89 @@
+package interpolatorwriter
+
+import (
+	"fmt"
+	"github.com/Azure/acs-engine/pkg/interpolator"
+	"io/ioutil"
+	"os"
+	"path"
+	"sync"
+)
+
+type InterpolatorWriter struct {
+	outputDirectory string
+	templateName    string
+	parametersName  string
+	interpolator    interpolator.Interpolator
+	otherFiles      map[string][]byte
+}
+
+func NewInterpolatorWriter(outputDirectory, templateName, parametersName string, i interpolator.Interpolator) *InterpolatorWriter {
+	return &InterpolatorWriter{
+		outputDirectory: outputDirectory,
+		templateName:    templateName,
+		parametersName:  parametersName,
+		interpolator:    i,
+		otherFiles:      make(map[string][]byte),
+	}
+}
+
+func (i *InterpolatorWriter) AddFile(name string, buffer []byte) {
+	i.otherFiles[name] = buffer
+}
+
+var InterpolatorWriterMutex sync.Mutex
+
+func (i *InterpolatorWriter) Write() error {
```

Comment:
I am wondering if we should make this implement `io.writer` I kind of feel like we should